File: Python/KthLargestElementInanArray.py
from typing import List
class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        min_value = min(nums)
        max_value = max(nums)

        count = [0] * (max_value - min_value + 1)

        for num in nums:
            count[num - min_value] += 1
        
        remaining = k
        for i in range(len(count) - 1, -1,-1):
            remaining -= count[i] 

            if remaining <= 0:
                return i + min_value
            
            
s=Solution()
nums=list(map(int,input("Array elements : ").split()))
k=int(input("k : "))
print(f"{k}th largest element in {nums} : {s.findKthLargest(nums,k)}")

File: LeetCode/Java/KthLargestElementInanArray.java
import java.util.Arrays;
import java.util.Scanner;

public class KthLargestElementInanArray {
    public int findKthLargest(int[] nums, int k) {
        int minValue = Integer.MAX_VALUE;
        int maxValue = Integer.MIN_VALUE;

        for (int num : nums) {
            minValue = Math.min(minValue, num);
            maxValue = Math.max(maxValue, num);
        }

        int[] count = new int[maxValue - minValue + 1];
        for (int num : nums) {
            count[num - minValue]++;
        }

        int remaining = k;
        for (int i = count.length - 1; i >= 0; i--) {
            remaining -= count[i];
            if (remaining <= 0) {
                return i + minValue;
            }
        }

        return -1;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        KthLargestElementInanArray s = new KthLargestElementInanArray();

        System.out.print("Array elements : ");
        String[] input = sc.nextLine().split(" ");
        int[] nums = Arrays.stream(input).mapToInt(Integer::parseInt).toArray();

        System.out.print("k : ");
        int k = sc.nextInt();

        int result = s.findKthLargest(nums, k);
        System.out.println(k + "th largest element in " + Arrays.toString(nums) + " : " + result);
    }
}
